graphql/resolvers.go

```diff
@@ -244,6 +246,84 @@ func (r *mutationResolver) SpawnHost(ctx context.Context, spawnHostInput *SpawnH
 	return &apiHost, nil
 }
 
+func (r *mutationResolver) UpdateSpawnHostStatus(ctx context.Context, hostID string, action string) (*restModel.APIHost, error) {
+	host, err := host.FindOneByIdOrTag(hostID)
+	if err != nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("Error finding host by id: %s", err))
+	}
+	usr := route.MustHaveUser(ctx)
+	env := evergreen.GetEnvironment()
+	queue := env.RemoteQueue()
+
+	if usr.Username() != host.StartedBy {
+		if !usr.HasPermission(gimlet.PermissionOpts{
+			Resource:      host.Distro.Id,
+			ResourceType:  evergreen.DistroResourceType,
+			Permission:    evergreen.PermissionHosts,
+			RequiredLevel: evergreen.HostsEdit.Value,
+		}) {
+			return nil, Forbidden.Send(ctx, "You are not authorized to modify this host")
+		}
+	}
+	var (
+		hostTerminate = "terminate"
+		hostStop      = "stop"
+		hostStart     = "start"
+	)
+	switch action {
+	case hostStart:
```

Comment:
Instead of copying the status change ones, can you put them in separate utility functions that are called by the old and new code?